parquet-column/src/main/java/org/apache/parquet/example/data/simple/SimpleGroup.java

```diff
@@ -48,28 +48,30 @@ public String toString() {
   }
 
   public String toString(String indent) {
-    String result = "";
+    StringBuilder result = new StringBuilder();
     int i = 0;
     for (Type field : schema.getFields()) {
       String name = field.getName();
       List<Object> values = data[i];
       ++i;
-      if (values != null) {
-        if (values.size() > 0) {
+      if (values != null && !values.isEmpty()) {
           for (Object value : values) {
-            result += indent + name;
+            result.append(indent);
+            result.append(name);
             if (value == null) {
-              result += ": NULL\n";
+              result.append(": NULL\n");
             } else if (value instanceof Group) {
-              result += "\n" + ((SimpleGroup)value).toString(indent+"  ");
+              result.append("\n");
+              result.append(((SimpleGroup)value).toString(indent+"  "));
```

Comment:
As we are doing this change because of performance reasons I would create a private method in `SimpleGroup` that gets the `StringBuilder` object is it can append all the stuff into the same builder.

Reply:
Isn't it already using same StringBuffer?

Reply:
This method is recursive in case of the child is a `Group`. This method appends the string generated by this recursive invocation. So, if you have a deeply nested structure it would create a bunch of `StringBuilder` objects, generate `String` objects from them and appends them to the `StringBuilder` object created at the higher level.
Instead, I would suggest passing the `StringBuilder` object created at the top level to the children's methods so you'll have only one builder and everything will be appended to it directly.

Reply:
Thanks for detail explanation, I didn't take a detail look in recursive. I think changing a method signature and using `StringBuilder `should be enough.